### ISSUE TITLE
Libtorrent takes up 100% in iotop, while only writing at ~800 kb/s

### ISSUE THREAD

**Issue by mtlife** [trapexit/mergerfs#260]
I'm using libtorrent via rtorrent on Ubuntu 15.10. Tested it with the Ubuntu server 14.04 amd64 torrent. Write speed to a non mergerfs filesystem is about 10 mb/s at nothing noticable in iotop, while the write speed on mergerfs is only around 800 kb/s with spikes to 1,2 mb/s and even sometimes drops as low as 300 kb/s. 

With direct_io enabled, libtorrent even refuses to write any data. Everything else seems to work fine, tried the following tests from this [link](https://www.thomas-krenn.com/en/wiki/Linux_I/O_Performance_Tests_using_dd) and got about 80 mb/s write speed without direct_io and 100 mb/s with it. Should be more than enough for libtorrent to also write at a higher speed.

I also found this link in my research: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=422432
But it seems that should already be solved long ago.

I've also found this link, tempered with the rtorrent settings to no avail on my system: http://forums.rutorrent.org/index.php?topic=1020.0

fstab looks like this:
```
# /etc/fstab: static file system information.
#
# Use 'blkid' to print the universally unique identifier for a
# device; this may be used with UUID= as a more robust way to name devices
# that works even if disks are added and removed. See fstab(5).
#
# <file system> <mount point>   <type>  <options>       <dump>  <pass>
# / was on /dev/sda1 during installation
UUID=uuid /               ext4    errors=remount-ro 0       1
# swap was on /dev/sda5 during installation
UUID=uuid none            swap    sw              0       0
# storage was added /dev/md/storage
UUID=uuid /mnt/storage  ext4    defaults 0 2
# and mergerfs was created
/mnt/storage:/old_home/ /home                     fuse.mergerfs defaults,category.create=eplfs,allow_other,minfreespace=10G,fsname=mergerfsdrv 0 0
```

And mergerfs -v:
```
mergerfs version: 2.12.3
FUSE library version: 2.9.4
fusermount version: 2.9.4
using FUSE kernel interface version 7.19
```

And uname a:
```
Linux ubuntu 4.2.0-30-generic #36-Ubuntu SMP Fri Feb 26 00:58:07 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

Edit: I forgot to mention that the root filesystem, as well as /mnt/storage are both mdadm raid1 setups. Not of any influence I think, I tested rtorrent to write directly to /mnt/storage without any io problems.

**Comment by trapexit** [comment]
Would you be able to log several seconds of an strace of rtorrent? I use Transmission and don't notice any issues. `strace -f -T -o /tmp/rtorrent.strace.txt -p <pid-of-rtorrent>`.

**Comment by mtlife** [comment]
Hi, I did it. The file is roughly a 415kb gzipped. I send it to you by mail.

**Comment by trapexit** [comment]
I don't see anything too odd. Perhaps a trace of mergerfs?

**Comment by mtlife** [comment]
Send another file to you.
![iotop rtorrent](https://cloud.githubusercontent.com/assets/1745548/13731624/1250dbbe-e96e-11e5-9b99-0399c50ac556.png)

And maybe some iostat output.
``` 
Linux 4.2.0-30-generic (ubuntu) 	03/13/16 	_x86_64_	(4 CPU)

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.21    0.52    1.10   29.08    0.00   67.09

Device:            tps    kB_read/s    kB_wrtn/s    kB_read    kB_wrtn
sda               3.70        15.27       521.48     466063   15912480
sdb               3.53         0.13       521.48       4078   15912480
sdc               4.04        46.00        25.20    1403703     768896
md127             5.20        15.33       521.34     467877   15908364
sdd               3.55        17.18        25.20     524266     768896
md126             4.77        59.35        15.34    1810901     468180
md125             3.32         3.75         9.53     114508     290920
```

**Comment by trapexit** [comment]
Turns out this is a kernel bug. 

https://lkml.org/lkml/2016/3/30/676

**Comment by mtlife** [comment]
https://lkml.org/lkml/2016/3/16/260
Isn't this the proper link?

**Comment by trapexit** [comment]
It's the same thread.

**Comment by mtlife** [comment]
To temporary fix it I installed the following kernel from vivid into wily without any issues so far.

    http://kernel.ubuntu.com/~kernel-ppa/mainline/v3.19.8-vivid/

You'll also need to install module-init-tools if you are on a fresh wily installation.

    apt-get install module-init-tools

**Comment by mtlife** [comment]
As posted by https://lkml.org/lkml/2016/5/11/59 

Can confirm this is fixed in kernel 4.4.10, which (for now). Should come to Ubuntu 16.04 soon. Or install it manually from:

    http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.4.10-xenial/
